src/BlockTypes/ProductButton.php

```diff
@@ -89,8 +89,10 @@ protected function render( $attributes, $content, $block ) {
 			$ajax_add_to_cart_enabled      = get_option( 'woocommerce_enable_ajax_add_to_cart' ) === 'yes';
 			$is_ajax_button                = $ajax_add_to_cart_enabled && ! $cart_redirect_after_add && $product->supports( 'ajax_add_to_cart' ) && $product->is_purchasable() && $product->is_in_stock();
 			$html_element                  = $is_ajax_button ? 'button' : 'a';
-			$styles_and_classes            = StyleAttributesUtils::get_classes_and_styles_by_attributes( $attributes, array( 'border_radius', 'font_size', 'font_weight', 'margin', 'padding', 'text_color' ) );
+			$styles_and_classes            = StyleAttributesUtils::get_classes_and_styles_by_attributes( $attributes );
 			$text_align_styles_and_classes = StyleAttributesUtils::get_text_align_class_and_style( $attributes );
+			$classname                     = isset( $attributes['className'] ) ? $attributes['className'] : '';
+			$custom_width_classes          = isset( $attributes['width'] ) ? 'has-custom-width wp-block-button__width-' . $attributes['width'] : '';
```

Comment:
What do you think about extending the `get_classes_and_styles_by_attributes` method with the `custom_width_classes` so it's there by default?

I did that for `text-align` attribute in this PR:
https://github.com/woocommerce/woocommerce-blocks/pull/8248/files#diff-bf7a423d751aba56975f9ca86523d396e462e59edf9d9d572109cad8617356c8

Or at least move the CSS class extraction to the `StyleAttributesUtils` class so it's shared and available for other blocks as well?

Reply:
Hi Karol! Thanks for pointing this out 🙌🏻

I am not 100% sure about this. Because this is a custom implementation for Add to cart button for now. You can see the code [here](https://github.com/woocommerce/woocommerce-blocks/blob/60c23add44c8bd2e2d17ebbf8f6f402fc94354ff/assets/js/atomic/blocks/product-elements/button/edit.tsx#L28-L72). Therefore, this functionality isn't shared with any other block for now.

But in future, If we will abstract this into a reusable block, then I agree with you that we will need to extend `get_classes_and_styles_by_attributes`.

What do you think?